runtime/doc/lua.txt

```diff
@@ -1536,6 +1536,27 @@ deepcopy({orig})                                              *vim.deepcopy()*
     Return: ~
         (table) Table of copied keys and (nested) values.
 
+defaulttable({create})                                    *vim.defaulttable()*
```

Comment:
we have `vim.empty_dict`. We should try to use consistent naming. Unless this also supports list-like tables, I guess `default_dict` is preferrable?

Reply:
As you wish, but we also have `tbl_*`, so the correct option is not clear to me.

I kind of prefer the `defaulttable` as it is closer to lua terms.

Reply:
`vim.empty_dict` is for use with the viml bridge as vim has dictionaries. Lua does not have dictionaries, it has tables so this is the naming we should use.

Reply:
@lewis6991 not super clear which name you're in favor of. This `defaulttable()` concept doesn't apply to list-like tables, so shouldn't we name it `defaultdict()` ?

Reply:
I think we could be better in general in the Lua API (especially `vim.tbl_*`) in distinguishing
* list-like (integer-indexed, may have holes)
* array-like (integer-indexed, must start at 1 and not have holes)
* dict-like (key-indexed)
tables.

May be worth a general cleanup PR.

Reply:
Pretty sure this does work for list-like tables.

```lua
A[1][1][1] = 1
```

Should work.

Reply:
Sorry, yes, I went from memory. (Forgot the definition in the Lua docs, which should take precedence.)

Thanks for linking the PR, which I meant to do.